packages/components/src/components/Tooltip/__tests__/Tooltip.ct.tsx

```diff
@@ -0,0 +1,321 @@
+            </Tooltip.Trigger>
+            <Tooltip.Content data-test-id={TOOLTIP_CONTENT_TEST_ID}>{TOOLTIP_TEXT}</Tooltip.Content>
+        </Tooltip.Root>,
+    );
+    const tooltipContent = page.getByTestId(TOOLTIP_CONTENT_TEST_ID);
+    await expect(component).toBeVisible();
+    await component.hover();
+    await expect(tooltipContent).toBeVisible();
+    const tooltipTriggerRect = await component.boundingBox();
+    const tooltipContentRect = await tooltipContent.boundingBox();
+    expect(tooltipContentRect).not.toBeUndefined();
+    expect(tooltipTriggerRect).not.toBeUndefined();
+    if (tooltipContentRect && tooltipTriggerRect) {
+        expect(tooltipContentRect.y + tooltipContentRect.height).toBeLessThan(tooltipTriggerRect.y);
+    }
+});
+
+test('should show on custom side top', async ({ mount, page }) => {
+    const component = await mount(
+        <Tooltip.Root>
+            <Tooltip.Trigger data-test-id={TOOLTIP_TRIGGER_TEST_ID}>
+                <button style={{ margin: 250 }}>Hover over me!</button>
+            </Tooltip.Trigger>
+            <Tooltip.Content side="top" data-test-id={TOOLTIP_CONTENT_TEST_ID}>
+                {TOOLTIP_TEXT}
+            </Tooltip.Content>
+        </Tooltip.Root>,
+    );
+    const tooltipContent = page.getByTestId(TOOLTIP_CONTENT_TEST_ID);
+    await expect(component).toBeVisible();
+    await component.hover();
+    await expect(tooltipContent).toBeVisible();
+    const tooltipTriggerRect = await component.boundingBox();
+    const tooltipContentRect = await tooltipContent.boundingBox();
+    expect(tooltipContentRect).not.toBeUndefined();
+    expect(tooltipTriggerRect).not.toBeUndefined();
+    if (tooltipContentRect && tooltipTriggerRect) {
+        expect(tooltipContentRect.y + tooltipContentRect.height).toBeLessThan(tooltipTriggerRect.y);
+    }
+});
+
+test('should show on custom side bottom', async ({ mount, page }) => {
+    const component = await mount(
+        <Tooltip.Root>
+            <Tooltip.Trigger data-test-id={TOOLTIP_TRIGGER_TEST_ID}>
+                <button style={{ margin: 250 }}>Hover over me!</button>
+            </Tooltip.Trigger>
+            <Tooltip.Content side="bottom" data-test-id={TOOLTIP_CONTENT_TEST_ID}>
+                {TOOLTIP_TEXT}
+            </Tooltip.Content>
+        </Tooltip.Root>,
+    );
+    const tooltipContent = page.getByTestId(TOOLTIP_CONTENT_TEST_ID);
+    await expect(component).toBeVisible();
+    await component.hover();
+    await expect(tooltipContent).toBeVisible();
+    const tooltipTriggerRect = await component.boundingBox();
+    const tooltipContentRect = await tooltipContent.boundingBox();
+    expect(tooltipContentRect).not.toBeUndefined();
+    expect(tooltipTriggerRect).not.toBeUndefined();
+    if (tooltipContentRect && tooltipTriggerRect) {
+        expect(tooltipTriggerRect.y + tooltipTriggerRect.height).toBeLessThan(tooltipContentRect.y);
+    }
```

Comment:
else throw (same for the other tests cases)

Reply:
adjusted